connector/authproxy/authproxy.go

```diff
@@ -47,11 +48,21 @@ func (m *callback) HandleCallback(s connector.Scopes, r *http.Request) (connecto
 	if remoteUser == "" {
 		return connector.Identity{}, fmt.Errorf("required HTTP header X-Remote-User is not set")
 	}
-	// TODO: add support for X-Remote-Group, see
-	// https://kubernetes.io/docs/admin/authentication/#authenticating-proxy
+	remoteGroups, ok := r.Header["X-Remote-Group"]
+	if !ok {
+		remoteGroups = []string{}
+	}
+	if strings.Contains(remoteUser, "@") {
+		remoteUserName := strings.SplitN(remoteUser, "@", 2)[0]
+		return connector.Identity{
+			Username:      remoteUserName,
+			Email:         remoteUser,
+			EmailVerified: true,
+			Groups:        remoteGroups,
+		}, nil
+	}
 	return connector.Identity{
-		UserID:        remoteUser, // TODO: figure out if this is a bad ID value.
-		Email:         remoteUser,
-		EmailVerified: true,
+		Username: remoteUser,
```

Comment:
So now we return fewer fields here if there are no groups?

Reply:
Groups could be empty list.
The UserID, Email and EmailVerified fields won't be set, if there was no '@' in the remoteUser.

Reply:
However the UserID is not set in any case.
Does UserID have higher precedence than Username, should we set it in all the cases?

Reply:
I think it has, but not sure!

The only thing I'm sure that we shouldn't break compatibility 🙂

Reply:
From what it's worse, I think not being too smart about the `email` field, and leaving it empty is probably a better option.

We don't know that the `X-Remote-User` contains an email and we probably shouldn't assume that unless the HTTP header had some precise email semantic (`X-Remote-Email` or something).

Reply:
Ok, I remove the Email and EmailVerified fields.
Can I set Username and UserID to the same value (remoteUser)?